doc/developer_guide/index.md

```diff
@@ -209,6 +247,7 @@ You will likely want to check out the
 
 - [Dev version of Panel Site](https://holoviz-dev.github.io/panel)
   - Use this to explore new, not yet released features and docs
+  - See which branch is currently deployed [here](https://github.com/holoviz/panel/actions/workflows/docs.yaml)
```

Comment:
You need to know what you are looking for to see what branch is deployed as the docs can be built in a dry run option.